Open_Foam/Domain.ipynb

```diff
@@ -79,11 +78,48 @@
     "    arc 2 5 (22.6274 22.6274 0)\n",
     "    arc 7 10 (0.3535 0.3535 1)\n",
     "    arc 8 11 (22.6274 22.6274 1)\n",
-    ");</pre>\n",
+    ");\n",
+    "```\n",
+    "\n",
+    "At this stage the `blockMeshDict` file is completely setup for creating the geometry of mesh of half of the CFD domain. User should save the `blockMeshDict` file and compile it to by running the following command in the working directory:\n",
+    "\n",
+    "```C++\n",
+    "blockMesh\n",
+    "```\n",
+    "\n",
+    "The output of compiling the created `blockmesh` file is visualized in Fig. 4.\n",
+    "\n",
+    "<img src=\"./Images/4halfMesh.png\" width=\"500\">\n",
+    "Fig. 4. The half CFD domain geometry and mesh produced by `blockMesh` utility.\n",
+    "\n",
+    "Now it is time to finish the creation of CFD domain's geometry and mesh by using the mirror boolean operation. This step is done by executing `mirrorMesh` utility. `mirrorMesh` would reflect the existing geometry and mesh with respect to a defined plane of symmetry. The settings for this command are defined in the `mirrorMeshDict` file located in constant/polyMesh directory as follows:\n",
+    "\n",
+    "```C++\n",
+    "planeType       pointAndNormal;\n",
+    "\n",
+    "pointAndNormalDict\n",
+    "{\n",
+    "    basePoint       (0 0 0);\n",
+    "    normalVector    (0 -1 0);\n",
+    "}\n",
+    "\n",
+    "planeTolerance  1e-06;\n",
+    "```\n",
     "\n",
-    "Finally its time to define the boundaries of the domain. The domain has five boundaries including: inlet, outlet, cylinder, symm and frontAndBack. Boundaries are face type geometries which are defined by the same vertices used in the blocking section. Each face is constructed by four vertices. A boundary can have one or more faces. In this section the basic boundary condition of each boundary is also defined. The vertices of a face are ordered as such that the normal vector of the face points to the inside of the domain.\n",
+    "The final step would be running the `mirrorMesh` command as follows:\n",
     "\n",
-    "<pre>boundary\n",
+    "```C++\n",
+    "mirrorMesh -overwrite\n",
+    "```\n",
+    "The `overwrite` flag in the command requires OpenFOAM to replace the previously saved mesh of the half domain mesh by the full domain mesh. Fig. 5 visualizes the full geometry and mesh created based on the created half domain and mesh using `mirrorMesh` command.\n",
+    "\n",
+    "<img src=\"./Images/5finalMesh.png\" width=\"500\">\n",
+    "Fig. 5. The final mesh  created based on the created half domain using the mirrorMesh utility.\n",
+    "\n",
```

Comment:
Well actually this is not right. The boundary file is file created after the meshing process. The file is considered as OpenFOAM mesh format in the polyMesh directory, but not a boundary defining section. But you may change it after the creation of the mesh.
To be more clear, blockMesh can play the role of GAMBIT and openFOAM can play the role of fluent. in GAMBIT we set boundary conditions at the last step just like blockMesh. Then we read the mesh inside the fluent and use it. The mesh format in fluent in invisible from the sights of the user. But in OpenFOAM after producing the mesh (Either by using blockMesh, snappyHexMesh, reading in from GAMBIT, SALOME and ...) The mesh is converted into the OpenFOAM mesh format which is stored in the polyMesh directory. One of the files is named boundary which consists boundary names and their respective face start number, face counts and the basic BC. One may change the BC name and BC types later in OpenFOAM by editing this file.